### PR TITLE
sink(ticdc): support multi statements in mysql backend

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: ref #8057

### What is changed and how it works?

- Support execute multi statements SQL in MySQL backend
- Add a parameter in `sink-uri` to control whether multi statements feature is enabled.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
Support multi statements SQL in mysql backend
```

### CODE REVIEW

File: pkg/sink/mysql/db_helper.go [hunk 1]
```diff
@@ -168,6 +168,7 @@ func generateDSNByConfig(
 		// set the `tidb_enable_external_ts_read` to `OFF`, so cdc could write to the sink
 		dsnCfg.Params["tidb_enable_external_ts_read"] = fmt.Sprintf(`"%s"`, tidbEnableExternalTSRead)
 	}
+
```
Comment (amyangfei) [inline pkg/sink/mysql/db_helper.go:171]:
```suggestion
```

File: cdc/sink/dmlsink/txn/mysql/mysql.go [hunk 2]
```diff
@@ -571,6 +571,83 @@ func (s *mysqlBackend) prepareDMLs() *preparedDMLs {
...
+			start, s.changefeed, multiStmtSQL, dmls.rowCount, dmls.startTs)
+		if rbErr := tx.Rollback(); rbErr != nil {
+			if errors.Cause(rbErr) != context.Canceled {
+				log.Warn("failed to rollback txn", zap.Error(rbErr))
+			}
+		}
+		cancelFunc()
+		return err
+	}
+	cancelFunc()
```
Comment (nongfushanquan) [inline cdc/sink/dmlsink/txn/mysql/mysql.go:602]: why call cancelFunc() here ?

Comment (amyangfei) [reply cdc/sink/dmlsink/txn/mysql/mysql.go]: use `defer cancel()` now

**Review by nongfushanquan — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Pull request merged**